### PR TITLE
Move core-generated files (config + saves) to frontend-defined save directory - 2nd attempt

### PR DESCRIPTION
**Pull request by jdgleaver**
Again, really sorry for making a hash of this the first time...

### CODE REVIEW

**Comment by jdgleaver** [comment]
Thanks for the warning! I can confirm this.

There is definitely something strange going on here...

- Under both Linux and Windows, the updated core loads **all** content correctly.

- Under Android, only the full version of the base game + the 2 official expansions work correctly. Attempting to run any unofficial expansion or mod causes the base game to load. It is as though the internal '-game' command line option is being ignored completely. And as stated, the 'content downloader' shareware version fails to load at all.

thatman84 - do you happen to know if the last version of the Android core was able to load 'unofficial' content correctly? I know this used to work, but the previous version of the core I was using was from before the buildbot switched to clang (and before the Android jni makefiles were updated) - i.e. I don't know if this content loading problem was an existing issue caused by other factors...

I can't do anything today (I'm not at home and don't have my stuff...), but I'll see if I can try to debug this next week...

**Comment by thatman84** [comment]
I will check tonight when im home.

I have the previous verstion on my shield tv most likely. Will test and grab the build number. I only updated the core when i started the forum topic not sure if that was before or after the clang thing

**Comment by thatman84** [comment]
Ok the core the only other core version for Android I can test is in this image

retroarch_20180422_194742

Shield TV
Killer Quake Mod tested and Quake(Shareware) from content downloader.

Hope that helps

**Comment by jdgleaver** [comment]
Thanks for this. The working core you have is from the commit before the build system changed. This makes me suspicious... I shall investigate!

**Comment by jdgleaver** [comment]
An update - I forgot to mention that this is now being tracked as issue #61.

TLDR version: I have identified the bug and found a fix, but I haven't a clue why the bug exists. I don't want to submit the fix if I don't know why it works, so I'm hoping someone else can help with the investigation...
